### PR TITLE
Updated auth

### PR DESCRIPTION
**Pull request by jc064522**

### CODE REVIEW

File: stroom-stats-service/src/main/java/stroom/stats/service/startup/App.java [hunk 1]
```diff
@@ -83,12 +81,19 @@ public void initialize(Bootstrap<Config> bootstrap) {
 
     @Override
     public void run(Config config, Environment environment) throws UnsupportedEncodingException {
-        configureAuthentication(config, environment);
-
         injector = Guice.createInjector(new StroomStatsServiceModule(config, hibernateBundle.getSessionFactory()));
         injector.getInstance(ServiceDiscoveryManager.class);
 
-        registerResources(environment);
+        if(config.getLogRequestsAndResponses()) {
```
Comment (at055612) [inline stroom-stats-service/src/main/java/stroom/stats/service/startup/App.java]: isLogRequestsAndResponsesEnabled ?
Think jackson supports is..... for booleans

File: stroom-stats-service/config.yml [hunk 2]
```diff
@@ -40,25 +40,38 @@ database:
     checkConnectionOnConnect: true
 
 
+auth:
```
Comment (at055612) [inline stroom-stats-service/config.yml]: unless these are needed immediately on app startup then they ought to be in zk with the rest of the config

File: stroom-stats-service/src/main/java/stroom/stats/service/startup/App.java (hunk 1 above)
Comment (jc064522) [reply stroom-stats-service/src/main/java/stroom/stats/service/startup/App.java]: Not sure 'is' is much better than 'get'. I wanted 'should'. I guess it's more canonical.

File: stroom-stats-service/config.yml [hunk 3]
```diff
@@ -99,62 +105,87 @@ zookeeper:
...
+    #----------HBase & related config START-----------------
     stroom.stats.hbase.config.client.write.buffer: 4194304
     stroom.stats.hbase.config.rpc.timeout: 180000
     stroom.stats.hbase.config.zookeeper.property.clientPort: ${STATS_HBASE_ZK_PORT:- 2181}
     #Comma delimited list of hosts in the ZK quorum, e.g. host1,host2,host3
     stroom.stats.hbase.config.zookeeper.quorum: ${STATS_HBASE_ZK_QUORUM:- localhost}
     #The znode in zookeeper for hbase to use as its root
     stroom.stats.hbase.config.zookeeper.znode.parent: ${STATS_HBASE_ZK_ZNODE_PARENT:- /hbase}
-    stroom.stats.hbase.dataStore.maxCheckAndPutRetries: 5
-    stroom.stats.hbase.dataStore.purge.intervalsToRetain.forever: 1
```
Comment (at055612) [inline stroom-stats-service/config.yml]: think this one got accidentally deleted

File: stroom-stats-service/src/main/java/stroom/stats/service/resources/query/v2/QueryResource.java [hunk 4]
```diff
@@ -119,15 +130,18 @@ public Response getDataSource(@Valid final DocRef docRef) {
...
-                searchRequest.getQuery().getDataSource(),
-                () -> Response.ok(hBaseClient.query(searchRequest)).build());
+    public Response search(
+            @Auth User user,
+            @NotNull @Valid SearchRequest searchRequest) {
+        LOGGER.debug("Received search request");
+
+        return Response.ok(hBaseClient.query(searchRequest)).build();
+
+        // TODO: Re-enable authorisations
```
Comment (at055612) [inline stroom-stats-service/src/main/java/stroom/stats/service/resources/query/v2/QueryResource.java]: does this need to be done?

File: stroom-stats-service/config.yml (hunk 3 above)
Comment (jc064522) [reply stroom-stats-service/config.yml]: Looks like they're still there but have been changed? Looks like this now I think:
```
+    #The minimum size of an aggregator to cause a flush to HBase
-    stroom.stats.aggregation.maxFlushIntervalMs: 25000
+    stroom.stats.aggregation.minBatchSize: 500000
+    #The maximum age of an aggregator before a flush to HBase is required
+    stroom.stats.aggregation.maxFlushIntervalMs: 20000
```

Comment (jc064522) [reply stroom-stats-service/config.yml]: I think they're there but the values have been changed?

**Pull request merged**